packages/gestalt/src/Sheet.js

```diff
@@ -8,33 +8,64 @@ type NodeOrRenderProp = Node | (({| onDismissStart: () => void |}) => Node);
    */
   accessibilityDismissButtonLabel?: string,
   /**
-   * Supply a short, descriptive label for screen-readers to contextualize the purpose of Sheet. See the [Accessibility section](#Accessibility) for more info.
+   * Supply a short, descriptive label for screen-readers to contextualize the purpose of Sheet. See the [Accessibility section](https://gestalt.pinterest.systems/web/sheet#Accessibility) for more info.
    */
   accessibilitySheetLabel: string,
   /**
-   * Supply the container element(s) or render prop that will be used as Sheet's main content. See the [animation variant](#Animation) for info on how to add exit animations to Sheet content..
+   * Supply the container element(s) or render prop that will be used as Sheet's main content. See the [animation variant](https://gestalt.pinterest.systems/web/sheet#Animation) for info on how to add exit animations to Sheet content..
    */
   children: NodeOrRenderProp,
   /**
-   * Indicate whether clicking on the backdrop (gray area) outside of Sheet will automatically close it. See the [outside click variant](#Preventing-close-on-outside-click) for more info.
+   * Indicate whether clicking on the backdrop (gray area) outside of Sheet will automatically close it. See the [outside click variant](https://gestalt.pinterest.systems/web/sheet#Preventing-close-on-outside-click) for more info.
    */
   closeOnOutsideClick?: boolean,
   /**
-   * Supply the container element(s) or render prop that will be used as Sheet's custom footer. See the [footer variant](#Footer) for more info..
+   * Supply the container element(s) or render prop that will be used as Sheet's custom footer. See the [footer variant](https://gestalt.pinterest.systems/web/sheet#Footer) for more info..
    */
   footer?: NodeOrRenderProp,
   /**
-   * The text used for Sheet's heading. Be sure to localize this text. See the [heading variant](#Heading) for more info.
+   * The text used for Sheet's heading. Be sure to localize this text. See the [heading variant](https://gestalt.pinterest.systems/web/sheet#Heading) for more info.
    */
   heading?: string,
   /**
-   * Callback fired when the Sheet in/out animations end. See the [animation](#Animation) variant to learn more.
+   * Callback fired when the Sheet in/out animations end. See the [animation](https://gestalt.pinterest.systems/web/sheet#Animation) variant to learn more.
    */
   onAnimationEnd?: ({| animationState: 'in' | 'out' |}) => void,
+  /**
+   * When supplied, it will disable component-controlled dismiss actions (ESC key press, backdrop click, or built-in dismiss IconButtons) and launch a confirmation Popover next to the dismiss IconButton requesting user confirmation before proceding. See the [dismiss confirmation](https://gestalt.pinterest.systems/web/sheet#Dismiss-confirmation) variant to learn more.
+   */
+  dismissConfirmation?:
+    | boolean
```

Comment:
I don't really like the polymorphism of this prop…but I also can't think of a better solution. 😬 Would passing an empty object literal instead of `true` be better? Would it be more annoying? Less intuitive?

Reply:
Actually the more I think about it I think an empty object is a better solution. It'll work anyways…and it's communicating the same thing ("I'd like a dismiss confirmation") while also communicating "I don't care to override any of the default labels"

Reply:
Like I think the polymorphism is also unintuitive, so if we're deciding against two unintuitive solutions, the one that makes more sense from a type perspective seems better?

Reply:
DONE